### PR TITLE
feat!: adding Namespace to remaining endpoints

### PR DESCRIPTION
**Pull request by distractedm1nd**
It turns out the previous addition of Namespace to Submit wasn't enough to allow for a native celestia-node implementation of go-da. 

GetIDs and Commitment have been extended to use Namespaces for DA layers that support it. 

~~Note that this assumes that namespace would land in the ID, which I'm not so sure about for the following reason:
ID is supposed to represent what uniquely identifies a blob on the DA layer, but adding Namespace means ID represents _**the data you need**_ to fetch a Blob uniquely from the DA layer. Also, changing the ID means that retrofitting existing op-stack rollups will break.~~

~~Whlie porting op-stack back to the blob module though, I realized that this change would only be necessary for using go-da. The id being only (height, commitment) is sufficient for fetching the data because you always know the namespace and can pass it into the blob endpoints directly.~~

~~The alternative to making Namespace a part of ID is adding Namespace to Get and Validate as well.~~

The signature for Submit has been changed again, thanks to a good catch from @tuxcanfly during review. We got rid of SubmitOptions again and pass Namespace as the last parameter, as in the other methods.

Now all methods take an optional Namespace as the last parameter, to avoid needing to modify ID. 

## Summary by CodeRabbit

- **New Features**
  - Enhanced the functionality to support namespaces for ID retrieval and commitment creation.
- **Refactor**
  - Updated proxy client and server methods to handle the new namespace parameter.
- **Documentation**
  - Updated the `DA Interface` table in the README.md to reflect changes in method parameters and the addition of a new method.
- **Style**
  - Modified the markdown linting configuration to disable a specific rule for tables.

### CODE REVIEW

File: da.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ type DA interface {
 	Get(ctx context.Context, ids []ID) ([]Blob, error)
 
 	// GetIDs returns IDs of all Blobs located in DA at given height.
-	GetIDs(ctx context.Context, height uint64) ([]ID, error)
+	GetIDs(ctx context.Context, height uint64, namespace Namespace) ([]ID, error)
 
 	// Commit creates a Commitment for each given Blob.
-	Commit(ctx context.Context, blobs []Blob) ([]Commitment, error)
+	Commit(ctx context.Context, blobs []Blob, namespace Namespace) ([]Commitment, error)
```
Comment (tuxcanfly) [inline da.go:20]: minor: given that namespace needs to be passed to the above functions as the last param, should we modify Submit for consistency, e.g.:

```Submit(ctx context.Context, blobs []Blob, opts *SubmitOptions)``` to

```Submit(ctx context.Context, blobs []Blob, gasPrice float64, namespace Namespace)```

Comment (distractedm1nd) [reply da.go:20]: I would like this better. Option structs are generally bad devex

**Comment by distractedm1nd** [comment]
@nashqueue and I are currently discussing if Validate really needs namespace - currently it is needed as part of the API to verify proofs in node (see: getByCommitment, blob.Included, nmt.Proof.VerifyInclusion). Will update this comment with results

**Comment by nashqueue** [comment]
I would rather have get and validate have namespace than id getting namespace.

File: proxy/client.go [hunk 2]
```diff
@@ -75,9 +75,10 @@ func (c *Client) GetIDs(ctx context.Context, height uint64) ([]da.ID, error) {
 }
 
 // Commit creates a Commitment for each given Blob.
-func (c *Client) Commit(ctx context.Context, blobs []da.Blob) ([]da.Commitment, error) {
+func (c *Client) Commit(ctx context.Context, blobs []da.Blob, namespace da.Namespace) ([]da.Commitment, error) {
 	req := &pbda.CommitRequest{
-		Blobs: blobsDA2PB(blobs),
+		Blobs:     blobsDA2PB(blobs),
+		Namespace: &pbda.Namespace{Value: namespace},
```
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply proxy/client.go:82]: Namespace is optional

[the remark this answers is not in the document]
Comment (distractedm1nd) [reply proxy/client.go:82]: If its not needed, users will pass an empty byte array (implementors of the interface will just throw away the parameter)

**Review by tuxcanfly — APPROVED** [review]

**Review by nashqueue — APPROVED** [review]

File: test/dummy.go [hunk 3]
```diff
@@ -100,7 +100,7 @@ func (d *DummyDA) Commit(ctx context.Context, blobs []da.Blob) ([]da.Commitment,
 }
 
 // Submit stores blobs in DA layer.
-func (d *DummyDA) Submit(ctx context.Context, blobs []da.Blob, opts *da.SubmitOptions) ([]da.ID, []da.Proof, error) {
+func (d *DummyDA) Submit(ctx context.Context, blobs []da.Blob, gasPrice float64, namespace da.Namespace) ([]da.ID, []da.Proof, error) {
```
[the remark this answers is not in the document]
Comment (nashqueue) [reply test/dummy.go:103]: Can do that in the follow up

**Pull request merged**